### PR TITLE
V22F-629 Replace Task with Allowed Black Box Actions

### PR DESCRIPTION
**Pull request by VoidLeech**

### CODE REVIEW

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Recorder.cpp [hunk 1]
```diff
@@ -270,6 +270,20 @@ bool UpdateV0RecorderToV1(void* p_settingsHandle, filesystem::path& p_userRecord
     return true;
 }
 
+/// @brief Update a v1 recording to a v2 recording. This means:
+///  - Adding allowed black box actions
+/// @param p_settingsHandle Handle to the run settings file
+/// @param p_userRecordingFile Path to the user recordings file
+/// @param p_decisionsRecordingFile Path to the decision recordings file
+/// @param p_simulationFile Path to the simulation file
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Recorder.cpp:278]: You don't use these params, so no need to add them to your method

**Review by suppergerrie2 — CHANGES_REQUESTED** [review]

**Review by suppergerrie2 — APPROVED** [review]

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/MediatorTests.cpp [hunk 2]
```diff
@@ -92,44 +95,36 @@ TEST(MediatorTests, ReadFromFile)
     ASSERT_TRUE(SetupSingletonsFolder());
     MockMediator* mediator1 = MockMediator::GetInstance();
 
-    MockMediator::ClearInstance();
+    MockMediator::ClearInstance(false);  // Our next call should real the pointer to the mediator from a file.
```
Comment (KungFuDonkey) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/MediatorTests.cpp:98]: typo in comment 
```suggestion
    MockMediator::ClearInstance(false);  // Our next call should read the pointer to the mediator from a file.
```

**Review by KungFuDonkey — APPROVED** [review]

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/DecisionsTests.cpp [hunk 3]
```diff
@@ -38,13 +38,14 @@ void InitializeMediator()
 TEST(DecisionsTest, RunInterveneDecisions)
 {
     InitializeMediator();
+    tAllowedActions allowedActions = {true, true, true};
```
Comment (noutheesink) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/DecisionsTests.cpp:41]: You should also test with each of these values being false at least once. Perhaps use the combinatorial tests for this as well

**Review by noutheesink — CHANGES_REQUESTED** [review]

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/DecisionsTests.cpp (hunk 3 above)
Comment (VoidLeech) [reply speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/DecisionsTests.cpp:41]: The decisions are independent as far as I can tell, they're just all tested in the same test. The testing method for true and false also differs, retrieving value first and making sure it is unchanged. So I don't think I combinatorial tests are the best for this. I do agree I should add tests though.

Comment (VoidLeech) [reply speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/DecisionsTests.cpp]: You can disregard this, it is actually achievable quite nicely.

**Review by noutheesink — APPROVED** [review]

**Pull request merged**